lib/Backend/GlobOpt.cpp

```diff
@@ -17435,7 +17435,18 @@ GlobOpt::PreOptPeep(IR::Instr *instr)
         }
         RemoveCodeAfterNoFallthroughInstr(instr);
     }
-
+    if (this->func->HasFinally())
+    {
```

Comment:
Add a good comment here.  This is a weird hack unfortunately, and we need to document it well.